### PR TITLE
filetree/nvim-tree: Open on directory

### PR DESCRIPTION
**Pull request by DamitusThyYeetus123**
When nvim-tree is enabled, opening nvim with a directory opens nvim-tree in that directory, and cds to the directory.

## Sanity Checking

[editorconfig]: https://editorconfig.org
[changelog]: https://github.com/NotAShelf/nvf/tree/main/docs/release-notes

- [ ] I have updated the [changelog] as per my changes.
- [x] I have tested, and self-reviewed my code.
- Style and consistency
  - [x] I ran **Alejandra** to format my code (`nix fmt`).
  - [x] My code conforms to the [editorconfig] configuration of the project.
  - [x] My changes are consistent with the rest of the codebase.
- If new changes are particularly complex:
  - [ ] My code includes comments in particularly complex areas
  - [ ] I have added a section in the manual.
  - [ ] _(For breaking changes)_ I have included a migration guide.
- Package(s) built:
  - [x] `.#nix` (default package)
  - [x] `.#maximal`
  - [ ] `.#docs-html`
- Tested on platform(s)
  - [x] `x86_64-linux`
  - [ ] `aarch64-linux`
  - [ ] `x86_64-darwin`
  - [ ] `aarch64-darwin`

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by NotAShelf** [comment]
Opening a directory with `vim .` should already open said directory with nvim-tree. I think this might be related to LazyLoading since we currently have a similar issue with neo-tree.

**Comment by diniamo** [comment]
#446 should fix this too.

**Comment by DamitusThyYeetus123** [comment]
This feature was removed from nvimTree and should be added externally as seen here https://github.com/nvim-tree/nvim-tree.lua/wiki/Open-At-Startup

**Comment by diniamo** [comment]
it's still going to have conflicts with that PR, so you have to at least rebase on it.

**Comment by DamitusThyYeetus123** [comment]
Is that PR not ready for commit yet?

**Comment by DamitusThyYeetus123** [comment]
#446 did not fix, will rebase and fix up now.

**Comment by DamitusThyYeetus123** [comment]
Rebased, no fixes needed

File: modules/plugins/filetree/nvimtree/config.nix [hunk 1]
```diff
@@ -77,22 +77,29 @@ in {
...
-              if not real_file and not no_name then
+              if not real_file and not directory and not no_name then
                 return
               end
 
               -- skip ignored filetypes
               if vim.tbl_contains(IGNORED_FT, filetype) then
                 return
-              end
+              end
```
Comment (horriblename) [inline modules/plugins/filetree/nvimtree/config.nix:97]:
```suggestion
              end
```

no space at end of line pls

**Review by horriblename — CHANGES_REQUESTED** [review]
LGTM, except the unnecessary space

**Comment by horriblename** [comment]
(waiting for CI to finish)

**Comment by DamitusThyYeetus123** [comment]
whoops lemme fix up formatting

File: modules/plugins/filetree/nvimtree/config.nix [hunk 2]
```diff
@@ -77,14 +77,17 @@ in {
               -- buffer is a real file on the disk
               local real_file = vim.fn.filereadable(data.file) == 1
 
+              -- buffer is a directory
+                local directory = vim.fn.isdirectory(data.file) == 1
```
Comment (NotAShelf) [inline modules/plugins/filetree/nvimtree/config.nix:81]:
```suggestion
              -- buffer is a directory
              local directory = vim.fn.isdirectory(data.file) == 1
```

**Review by horriblename — APPROVED** [review]

**Pull request merged**